domain/consensus/processes/difficultymanager/blockwindow.go

```diff
@@ -53,19 +55,20 @@ func (dm *difficultyManager) blockWindow(stagingArea *model.StagingArea, startin
 	return window, windowHashes, nil
 }
 
-func (window blockWindow) minMaxTimestamps() (min, max int64, minIndex, maxIndex int) {
+func (window blockWindow) minMaxTimestamps() (min, max int64, minIndex int) {
 	min = math.MaxInt64
-	minIndex = math.MaxInt64
+	minIndex = 0
 	max = 0
-	maxIndex = 0
 	for i, block := range window {
-		if block.timeInMilliseconds < min {
+
+		if block.timeInMilliseconds < min ||
+			(block.timeInMilliseconds == min && block.blueWork.Cmp(window[minIndex].blueWork) < 0) ||
+			(block.timeInMilliseconds == min && block.blueWork.Cmp(window[minIndex].blueWork) == 0 && block.hash.Less(window[minIndex].hash)) {
```

Comment:
Maybe split into `if block.timeInMilliseconds == min`, and maybe consider copying the full `Less` function

Reply:
Done